commerce/bookstore/schema.tql

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright (C) 2022 Vaticle
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+define
+
+### Attributes ###
+
+    name sub attribute, value string;
+    date sub attribute, value datetime;
```

Comment:
But if order is a relation and it connects a single ordered-item to a customer - then we can't have more than 1 item in each order instance. That's why I created Order as an entity. It exists if you think of this example as a CRM - order is the main entity. I understand it lacks functionality in this example, but storing a list of items, creation date, status, and delivery-address is enough IMHO. Or am I missing something here?

Reply:
Actually, much like attribute ownerships, **all** relation roleplayers are currently one-to-many as well!

So you can insert
```
insert $order (item: $i1, item: $i2, item: $i3, customer: $cust) isa order, has placed-date '09-12-2022';
```

An alternative model is to introduce a new relation, `basket`, where a `basket` is simply defined as `define basket sub relation, relates basket-content` and you again just insert a bunch of contents.

Having said that, this may be a weak model, because it's valid for an empty `basket` to exist, yet a relation with no roleplayers is (currently) considered very strange, and we're considering deprecating them in the future.

Reply:
> all relation roleplayers are currently one-to-many

> insert $order (item: $i1, item: $i2, item: $i3, customer: $cust) isa order, has placed-date '09-12-2022';

Oh, that's cool. Let's do that just to show this in the example.

Reply:
I have fixed it at https://github.com/vaticle/typedb-examples/pull/143/commits/88aeae70ac3a5a1b77edd85da6767b2c05b87568